src/lib/import_export/csv_parser.cpp

```diff
@@ -18,10 +19,19 @@
 
 namespace opossum {
 
-CsvParser::CsvParser(const CsvConfig& csv_config) : _csv_config(csv_config) {}
+CsvParser::CsvParser() {}
 
-std::shared_ptr<Table> CsvParser::parse(const std::string& filename) {
-  const auto table = _process_meta_file(filename + _csv_config.meta_file_extension);
+std::shared_ptr<Table> CsvParser::parse(const std::string& filename, const std::optional<CsvMeta> csv_meta) {
+  if (csv_meta == std::nullopt) {
+    _meta = process_csv_meta_file(filename + CsvMeta::META_FILE_EXTENSION);
+  } else {
+    _meta = csv_meta.value();
+  }
+  return _parse(filename);
+}
+
+std::shared_ptr<Table> CsvParser::_parse(const std::string& filename) {
+  const auto table = _create_table_from_meta();
```

Comment:
Argh. I really don't like the syntax of `const` when it comes to `shared_ptr`s. This looks like we have a const table, but instead we have a const pointer. :/

Reply:
Right.... Actually I don't know why I used it here anyway... I removed it.